astropy/utils/misc.py

```diff
@@ -382,18 +384,43 @@ class JsonCustomEncoder(json.JSONEncoder):
 
     """
     def default(self, obj):
+        from .. import units as u
         import numpy as np
-        if isinstance(obj, (np.ndarray, np.number)):
+        if isinstance(obj, np.number):
```

Comment:
Is there a reason why you didn't just put `Quantity` as the first check and retain `isinstance(obj, (np.ndarray, np.number))` as-is as the second check?